### PR TITLE
Fix safe browsing patches

### PR DESCRIPTION
**Pull request by tangalbert919**
A follow-up PR for #840.

This PR fixes whatever I've missed and other users have spotted as they attempted to build Chromium 78 with these patches.

### CODE REVIEW

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch [hunk 1]
```diff
@@ -73,26 +73,101 @@
...
+-      GetPasswordProtectionService();
+-  if (pps) {
+-    pps->MaybeStartPasswordFieldOnFocusRequest(
+-        web_contents(), web_contents()->GetLastCommittedURL(), form_action,
+-        frame_url, pps->GetAccountInfo().hosted_domain);
+-  }
+-}
+-#endif  // defined(ON_FOCUS_PING_ENABLED)
+-
+ #if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
```
Comment (Eloston) [inline patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch]: Do you know what sets `SYNC_PASSWORD_REUSE_DETECTION_ENABLED`? Is it a GN flag?

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch [hunk 2]
```diff
@@ -349,3 +424,57 @@
...
+ #include "components/prefs/pref_member.h"
+-#include "components/safe_browsing/buildflags.h"
+ #include "content/public/browser/render_widget_host.h"
+ #include "content/public/browser/web_contents_binding_set.h"
+ #include "content/public/browser/web_contents_observer.h"
+@@ -176,13 +175,6 @@ class ChromePasswordManagerClient
+           saving_flow_recorder);
+ #endif  // defined(OS_ANDROID)
+ 
+-#if defined(ON_FOCUS_PING_ENABLED)
```
Comment (Eloston) [inline patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch]: I'm also wondering if `ON_FOCUS_PING_ENABLED` is set by a GN flag too

File: patches/core/inox-patchset/0001-fix-building-without-safebrowsing.patch [hunk 3]
```diff
@@ -1170,3 +1170,65 @@
...
+ 
+--- a/chrome/browser/profiles/profiles_state.cc
++++ b/chrome/browser/profiles/profiles_state.cc
+@@ -212,12 +212,6 @@ bool SetActiveProfileToGuestIfLocked() {
+ #endif  // !defined(OS_CHROMEOS)
+ 
+ void RemoveBrowsingDataForProfile(const base::FilePath& profile_path) {
+-  // The BrowsingDataRemover relies on many objects that aren't created in unit
+-  // tests. Previously this code would depend on content::ResourceDispatcherHost
+-  // but that's gone, so do a similar hack for now.
```
Comment (Eloston) [inline patches/core/inox-patchset/0001-fix-building-without-safebrowsing.patch]: I'm missing the context on this comment. Do you know why their "hack" depends on checking for the Safe Browsing service before running the rest of the function?

**Comment by jstkdng** [comment]
Have you succesfully built with this patches? asking for a friend

**Comment by tangalbert919** [comment]
I have not built with these patches yet, but I am going to as soon as I can.

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch (hunk 1 above)
Comment (tangalbert919) [reply patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch]: `SYNC_PASSWORD_REUSE_DETECTION_ENABLED` is set by the following section of `components/password_manager/core/browser/BUILD.gn`:

```
# TODO(crbug.com/706392): Fix password reuse detection for Android.
password_reuse_detection_support = !is_android && !is_ios
password_on_focus_ping_support = !is_ios

config("password_reuse_detection_config") {
  defines = []
  if (password_reuse_detection_support) {
    defines += [ "SYNC_PASSWORD_REUSE_DETECTION_ENABLED" ]
  }
  if (password_on_focus_ping_support) {
    defines += [ "ON_FOCUS_PING_ENABLED" ]
  }
}
```

`ON_FOCUS_PING_ENABLED` is also set here as well.

**Comment by billyblackburn** [comment]
Have u seen the qutebrowser?Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Albert Tang <notifications@github.com> Date: 10/29/19  3:28 PM  (GMT-05:00) To: Eloston/ungoogled-chromium <ungoogled-chromium@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [Eloston/ungoogled-chromium] (WIP) Fix safe browsing patches (#843) @tangalbert919 commented on this pull request.



In patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch:
> -     const GURL& form_action,
-     const GURL& frame_url) {
+-void ChromePasswordManagerClient::CheckSafeBrowsingReputation(
+-    const GURL& form_action,
+-    const GURL& frame_url) {
+-  safe_browsing::PasswordProtectionService* pps =
+-      GetPasswordProtectionService();
+-  if (pps) {
+-    pps->MaybeStartPasswordFieldOnFocusRequest(
+-        web_contents(), web_contents()->GetLastCommittedURL(), form_action,
+-        frame_url, pps->GetAccountInfo().hosted_domain);
+-  }
+-}
+-#endif  // defined(ON_FOCUS_PING_ENABLED)
+-
+ #if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)

SYNC_PASSWORD_REUSE_DETECTION_ENABLED is set by the following section of components/password_manager/core/browser/BUILD.gn:
# TODO(crbug.com/706392): Fix password reuse detection for Android.
password_reuse_detection_support = !is_android && !is_ios
password_on_focus_ping_support = !is_ios

config("password_reuse_detection_config") {
  defines = []
  if (password_reuse_detection_support) {
    defines += [ "SYNC_PASSWORD_REUSE_DETECTION_ENABLED" ]
  }
  if (password_on_focus_ping_support) {
    defines += [ "ON_FOCUS_PING_ENABLED" ]
  }
}

ON_FOCUS_PING_ENABLED is also set here as well.

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/Eloston/ungoogled-chromium/pull/843?email_source=notifications\u0026email_token=ABYOVORDL5SWOXKV2EMF77DQRCFEDA5CNFSM4JGNJBM2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJTY5TQ#discussion_r340281431",
"url": "https://github.com/Eloston/ungoogled-chromium/pull/843?email_source=notifications\u0026email_token=ABYOVORDL5SWOXKV2EMF77DQRCFEDA5CNFSM4JGNJBM2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJTY5TQ#discussion_r340281431",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by jstkdng** [comment]
@billyblackburn yeah its a nice browser but it doesn't have a lot of extensions. You can get the shortcuts from qutebrowser into chromium by using the extension vimium. Also, qutebrowser lacks hardware acceleration afaik.

**Comment by tangalbert919** [comment]
I have attempted building with this set of patches and got no compile errors this time. Should be safe to merge.

**Comment by tangalbert919** [comment]
I'll push what I had to change on `ungoogled-chromium-archlinux` once this PR is merged.

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch (hunk 2 above)
Comment (tangalbert919) [reply patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch]: I addressed this question in the review above.

**Pull request merged**